cli/internal/run/run.go

```diff
@@ -402,165 +414,103 @@ type runOpts struct {
-					return nil, err
-				} else {
-					opts.runOpts.concurrency = concurrency
-				}
-			case strings.HasPrefix(arg, "--includeDependencies"):
-				output.Warn("[WARNING] The --includeDependencies flag has renamed to --include-dependencies for consistency. Please use `--include-dependencies` instead")
-				opts.scopeOpts.LegacyFilter.IncludeDependencies = true
-			case strings.HasPrefix(arg, "--include-dependencies"):
-				opts.scopeOpts.LegacyFilter.IncludeDependencies = true
-			case strings.HasPrefix(arg, "--only"):
-				opts.runOpts.only = true
-			case strings.HasPrefix(arg, "--output-logs="):
-				outputLogsMode := arg[len("--output-logs="):]
-				switch outputLogsMode {
-				case "full":
-					opts.runcacheOpts.CacheMissLogsMode = runcache.FullLogs
-					opts.runcacheOpts.CacheHitLogsMode = runcache.FullLogs
-				case "none":
-					opts.runcacheOpts.CacheMissLogsMode = runcache.NoLogs
-					opts.runcacheOpts.CacheHitLogsMode = runcache.NoLogs
-				case "hash-only":
-					opts.runcacheOpts.CacheMissLogsMode = runcache.HashLogs
-					opts.runcacheOpts.CacheHitLogsMode = runcache.HashLogs
-				case "new-only":
-					opts.runcacheOpts.CacheMissLogsMode = runcache.FullLogs
-					opts.runcacheOpts.CacheHitLogsMode = runcache.HashLogs
-				default:
-					output.Warn(fmt.Sprintf("[WARNING] unknown value %v for --output-logs CLI flag. Falling back to full", outputLogsMode))
-				}
-			case strings.HasPrefix(arg, "--dry-run"):
-				opts.runOpts.dryRun = true
-				if strings.HasPrefix(arg, "--dry-run=json") {
-					opts.runOpts.dryRunJSON = true
-				}
-			case strings.HasPrefix(arg, "--dry"):
-				opts.runOpts.dryRun = true
-				if strings.HasPrefix(arg, "--dry=json") {
-					opts.runOpts.dryRunJSON = true
-				}
-			case strings.HasPrefix(arg, "--remote-only"):
-				opts.cacheOpts.SkipFilesystem = true
-			case strings.HasPrefix(arg, "--team"):
-			case strings.HasPrefix(arg, "--token"):
-			case strings.HasPrefix(arg, "--preflight"):
-			case strings.HasPrefix(arg, "--api"):
-			case strings.HasPrefix(arg, "--url"):
-			case strings.HasPrefix(arg, "--trace"):
-			case strings.HasPrefix(arg, "--cpuprofile"):
-			case strings.HasPrefix(arg, "--heap"):
-			case strings.HasPrefix(arg, "--no-gc"):
-			case strings.HasPrefix(arg, "--cwd="):
-			default:
-				return nil, errors.New(fmt.Sprintf("unknown flag: %v", arg))
-			}
-		}
+func (d *dryRunValue) Set(value string) error {
+	if value == "json" {
+		d.opts.dryRun = true
+		d.opts.dryRunJSON = true
+	} else if value == "text|json" {
+		d.opts.dryRun = true
+	} else {
+		return fmt.Errorf("invalid dry-run mode: %v", value)
 	}
+	return nil
+}
 
-	// We can only set this cache folder after we know actual cwd
-	if unresolvedCacheFolder != "" {
-		opts.cacheOpts.Dir = fs.ResolveUnknownPath(config.Cwd, unresolvedCacheFolder)
-	}
-	if !config.IsLoggedIn() {
-		opts.cacheOpts.SkipRemote = true
-	}
+func (d *dryRunValue) Type() string {
```

Comment:
I have no idea what this is doing. Can you help me understand? (It doesn't appear in old code either.) Is it specifying the alias for some help text somewhere?

Reply:
The short answer is that this is required to implement the `pflag.Value` interface. 

The longer answer is that it is used for a variety of things internal to how flags are parsed and displayed in help/usage text. I'll add some comments, or maybe pull this out into its own utility.